src/__tests__/db.factory.ts

```diff
@@ -3,28 +3,57 @@ import fs from 'node:fs';
 import path from 'node:path';
 import configuration from '@/config/entities/__tests__/configuration';
 
-export function dbFactory(): postgres.Sql {
-  const config = configuration();
-  const isCIContext = process.env.CI?.toLowerCase() === 'true';
+export class TestDbFactory {
+  private readonly config = configuration();
+  private readonly isCIContext = process.env.CI?.toLowerCase() === 'true';
+  private readonly mainConnection: postgres.Sql;
+  private static readonly TEST_CERTIFICATE_PATH = path.join(
+    process.cwd(),
+    'db_config/test/server.crt',
+  );
```

Comment:
Nit: I'd move this to the top of the class due to the casing.

Reply:
Done [e09881a](https://github.com/safe-global/safe-client-gateway/pull/1735/commits/e09881aa7a589b485757217b3a1a9f8f26ae22f8)